includes/class-wc-purchased-products-helper.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * WC_Purchased_Products_Helper class.
+ *
+ * @class     WC_Purchased_Products_Helper
+ * @version   3.3.0
+ * @package   WooCommerce/Classes
+ * @category  Class
+ * @author    WooThemes
+ */
+class WC_Purchased_Products_Helper {
+
+	/**
+	 * Hook in methods.
+	 */
+	public static function init() {
+		add_action( 'woocommerce_delete_order', array( __CLASS__, 'handle_delete_order' ) );
+		add_action( 'woocommerce_before_order_object_save', array( __CLASS__, 'handle_save_order' ) );
```

Comment:
Spoke with @mikejolley about this. It's better to use `woocommerce_order_object_updated_props`, and also fixes the current todo because it happens after save.